src/lib_json/json_writer.cpp

```diff
@@ -141,14 +142,15 @@ std::string valueToString(double value, bool useSpecialFloats, unsigned int prec
   char buffer[32];
   int len = -1;
 
-  char formatString[6];
-  sprintf(formatString, "%%.%dg", precision);
+  std::stringstream value_ss;
+  value_ss << std::setprecision( (int)precision ) << value;
```

Comment:
This cast isn't necessary. The precision must have a value small enough to be representable as an int. If that condition holds, this is a well-defined conversion without the cast. If this condition does not hold, then then this cast is invalid. The cast doesn't provide any value as far as I can see. If you need it for some reason (compiler warnings?), it's better as static_cast<int>(precision), which is a weaker cast than C-style casting.